### PR TITLE
Wip/hughsie/meson

### PR DESCRIPTION
**Pull request by hughsie**

### CODE REVIEW

File: meson_options.txt [hunk 1]
```diff
@@ -0,0 +1,21 @@
+option('packaging_backend', type : 'combo', choices : ['alpm', 'aptcc', 'dnf', 'dummy', 'entropy', 'pisi', 'poldek', 'portage', 'ports', 'slack', 'urpmi', 'yum', 'zypp', 'nix'], value : 'dummy', description : 'The name of the backend to use')
+option('dnf_vendor', type : 'combo', choices : ['fedora', 'mageia', 'openmandriva'], value : 'fedora', description : 'Vendor configuration when using the dnf backend')
```
Comment (Conan-Kudo) [inline meson_options.txt]: We're missing the newly-added `rosa` value here.

Comment (hughsie) [reply meson_options.txt]: Fixed, thanks. Will push after the next travis build.

**Comment by ximion** [comment]
`-DPK_COMPILATION` could probably be set as global argument in the main Meson file, at least for the C language: `add_global_arguments('-DPK_COMPILATION', language : 'c')`
That will avoid a tiny bit of duplication.

**Comment by hughsie** [comment]
> DPK_COMPILATION could probably be set as global argument

Good idea, done.

File: backends/aptcc/meson.build [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    apt_pkg_dep,
+    apt_inst_dep,
+    gstreamer_dep,
+    gstreamer_base_dep,
+    gstreamer_plugins_base_dep,
+    appstream_dep,
+  ],
+  cpp_args: [
+    '-DG_LOG_DOMAIN="PackageKit-APTcc"',
+    '-DPK_COMPILATION=1',
```
Comment (Conan-Kudo) [inline backends/aptcc/meson.build]: We'll need this in the nix, slack, and zypp backends too.

**Comment by hughsie** [comment]
I'm tempted to merge this tomorrow, and we can work out any problems in master as fixups. Speak now, or forever hold your peace :)

**Comment by Conan-Kudo** [comment]
I'd say merge it now, so that this doesn't get more monstrous.

**Comment by ximion** [comment]
Merge tomorrow sounds fine :-)

**Pull request merged**